src/Mailer/Handler/SendMailsForOwnershipEventHandler.php

```diff
@@ -21,20 +21,26 @@ final class SendMailsForOwnershipEventHandler implements EventListener
     use AsyncDispatchTrait;
 
     private ContextDecoratedCommandBus $mailerCommandBus;
-    private DomainMessageSpecificationInterface $isReplay;
+
+    /**
+     * @var DomainMessageSpecificationInterface[]
+     */
+    private array $domainMessageSpecifications;
 
     public function __construct(
         ContextDecoratedCommandBus $mailerCommandBus,
-        DomainMessageSpecificationInterface $domainMessageSpecification
+        DomainMessageSpecificationInterface ...$domainMessageSpecification
```

Comment:
I'm not sure about this change, because now you could use the `SendMailsForOwnershipEventHandler` without passing the extra `DomainMessageHasMailsDisabled`. This could potentially lead to still sending emails.